tests/unit/scheduler/graphql_mid/test_graphql_subs.py

```diff
@@ -0,0 +1,87 @@
+import pytest
+
+from lucupy.observatory.abstract import ObservatoryProperties
+from lucupy.observatory.gemini import GeminiProperties
+
+from scheduler.graphql_mid.server import schema
+
+
+@pytest.mark.asyncio
+async def test_schedule_sub(visibility_calculator_fixture, set_observatory_properties):
+
+    query = """
+            query Schedule {
+                schedule(scheduleId: "1", 
+                         newScheduleInput: {startTime: "2018-10-01 08:00:00",
+                                            endTime: "2018-10-04 08:00:00"
+                                            sites: "GN", 
+                                            mode: VALIDATION,
+                                            semesterVisibility:false,
+                                            numNightsToSchedule:3})
+            }
+    """
+    sub = """
+         subscription QueueSchedule {
+          queueSchedule(scheduleId: "1") {
+            __typename
+            ... on NewNightPlans {
+              nightPlans {
+                nightTimeline {
+                  nightIndex
+                  timeEntriesBySite {
+                    site
+                    timeLosses
+                    timeEntries {
+                      event
+                      plan {
+                        startTime
+                        endTime
+                        nightConditions {
+                          iq
+                          cc
+                        }
+                        visits {
+                          obsId
+                        }
+                        nightStats {
+                          planScore
+                          timeLoss
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+              plansSummary
+            }
+            ... on NightPlansError {
+              error
+            }
+          }
+        } 
+    """
+    sub_response = await schema.subscribe(sub)
+    result = await schema.execute(query)
+
+    async for result in sub_response:
+        # Check return without errors
+        assert not result.errors, 'Subscription returned with errors'
+        # Check the correct number of nights.
+        print([result.data["queueSchedule"]])
+        n_nights = len(result.data["queueSchedule"]["nightPlans"]["nightTimeline"])
+        print('Result night timeline: ', result.data["queueSchedule"]["nightPlans"]["nightTimeline"])
+        assert n_nights == 1, f'Number of nights must be 3, but got {n_nights}'
```

Comment:
Should be 3 instead of 1 in the assertion.
`assert n_nights == 3, f'Number of nights must be 3, but got {n_nights}'`

Reply:
Thanks! Have to change the message!